src/lib/etasu.ts

```diff
@@ -80,11 +81,309 @@ router.post('/reset', async (req: Request, res: Response) => {
+  const remsRequest: any = {
+    case_number: case_number,
+    status: remsRequestCompletedStatus,
+    drugName: drug?.name,
+    drugCode: drug?.code,
+    patientFirstName: patientFirstName,
+    patientLastName: patientLastName,
+    patientDOB: patientDOB,
+    metRequirements: []
+  };
+
+  // create the metReq that was submitted
+  const metReq = {
+    completed: true,
+    completedQuestionnaire: questionnaireResponse,
+    requirementName: requirement.name,
+    requirementDescription: requirement.description,
+    drugName: drug?.name,
+    stakeholderId: reqStakeholderReference,
+    case_numbers: [case_number]
+  };
+
+  if (!(await createAndPushMetRequirements(metReq, remsRequest))) {
+    createNewCase = false;
+    res.status(200);
+    message = 'ERROR: failed to create new met requirement for form initial to case';
+    console.log(message);
+    res.send(message);
+    return res;
+  }
+
+  // iterate through all other reqs again to create corresponding false metReqs / assign to existing
+  for (const requirement2 of drug.requirements) {
+    // skip if the req found is the same as in the outer loop and has already been processed
+    // && If the requirement is not the patient Status Form (when requiredToDispense == false)
+    if (!(requirement2.resourceId === requirement.resourceId) && requirement2.requiredToDispense) {
+      // figure out which stakeholder the req corresponds to
+      const reqStakeholder2 = requirement2.stakeholderType;
+      const reqStakeholder2Reference =
+        reqStakeholder2 === 'prescriber'
+          ? practitionerReference
+          : reqStakeholder2 === 'pharmacist'
+          ? pharmacistReference
+          : patientReference;
+
+      const matchedMetReq2 = await metRequirementsCollection
+        .findOne({
+          stakeholderId: reqStakeholder2Reference,
+          requirementName: requirement2.name,
+          drugName: drug?.name
+        })
+        .exec();
+      if (matchedMetReq2) {
+        pushMetRequirements(matchedMetReq2, remsRequest);
+
+        if (!matchedMetReq2.completed) {
+          remsRequestCompletedStatus = 'Pending';
+        }
+        matchedMetReq2.case_numbers.push(case_number);
+        await matchedMetReq2.save();
+      } else {
+        // create the metReq that was submitted
+        const newMetReq = {
+          completed: false,
+          completedQuestionnaire: null,
+          requirementName: requirement2.name,
+          requirementDescription: requirement2.description,
+          drugName: drug?.name,
+          stakeholderId: reqStakeholder2Reference,
+          case_numbers: [case_number]
+        };
+
+        remsRequestCompletedStatus = 'Pending';
+
+        if (!(await createAndPushMetRequirements(newMetReq, remsRequest))) {
+          message = 'ERROR: failed to create new met requirement for form initial to case';
+          console.log(message);
+        }
+      }
+    }
+  }
+
+  if (createNewCase) {
```

Comment:
is this if needed? The only time this would be false is if there was an error on line 154, but at that point res.send() has also already been submitted so this line here at 213 won't be hit and neither would lines 218-219 since we already sent the response prior. Alternatively we could keep the if but lines 218-219 should be moved inside the if I believe. If somehow we hit this line and don't go into the if (which I don't think would happen), this code introduces the possibility of sending returnedRemsRequestDoc in the response body as undefined since the only place it gets set after instantiation is line 215 in this function (this could also potentially be made into a const since its only set in one place)

Reply:
minor code clean up